sdk/openai/Azure.AI.OpenAI.Assistants/src/Custom/RunStep.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI.Assistants;
+
+public partial class RunStep
+{
+    /*
+     * CUSTOM CODE DESCRIPTION:
+     *
+     * "Required but nullable" utcDateTime doesn't emit the appropriate deserialization logic by default.
```

Comment:
I haven't filed an issue because it falls outside of normal and supported usage -- we had to `#suppress` the warning in TypeSpec to make this work, after all. Do you believe the emitter should still appropriately handle it despite the warning suppression required? If so, happy to file it.

Reply:
Will follow up on treatment here separately.